lib/sign.js

```diff
@@ -17,11 +18,51 @@ var xpi = require("./xpi");
 
 var AMO_API_PREFIX = require("./settings").AMO_API_PREFIX;
 
+function getXpiInfo(options) {
+  options = _.assign({
+    addonDir: null,
+    xpiPath: null,
+  }, options);
+
+  if (!options.xpiPath) {
+    return utils.getManifest({addonDir: options.addonDir})
+      .then(function(manifest) {
+        return getXpiInfoFromManifest(manifest);
+      });
+  } else {
+    return utils.inspectXPI(options.xpiPath)
```

Comment:
Looks like this could be:

```js
return utils.getXPIManifest(options.xpiPath)
    .then(function(manifest) {
      if (manifest) {
        return getXpiInfoFromManifest(manifest);
      } else {
        throw new Error("XPIs without a package.json are not supported yet");
      }
    });